wpimath/src/main/java/edu/wpi/first/wpilibj/waypoint/WaypointUtil.java

```diff
@@ -0,0 +1,124 @@
+  }
+
+  /**
+   * Imports Path object and creates a trajectory with that info.
+   *
+   * @param filepath Full path to the file
+   * @param config TrajectoryConfig that sets the contraints for the Trajectory to be generated
+   *
+   * @return Trajectory created from the waypoints in the file.
+   */
+  public static Trajectory importPathToCubicTrajectory(String filepath, TrajectoryConfig config) throws IOException {
+    try (FileReader fr = new FileReader(new File(filepath));
+      BufferedReader reader = new BufferedReader(fr)) {
+      
+      var interiorWaypoints = new ArrayList<Translation2d>();
+      Pose2d start = new Pose2d();
+      Pose2d end = new Pose2d();
+      int loop = 0;
+      String line;
+      String lastline = "";
+
+      while ((line = reader.readLine()) != null) {
+        if (loop == 0) {
+          if (!line.equals("X,Y,Tangent X,Tangent Y,Fixed Theta,Reversed,Name")) {
+            throw new RuntimeException("this isn’t a PathWeaver csv file");
+          }  
+        }
+        else if (loop == 1) {
+          start = createPoseWaypoint(line);
+        }
+        else if (loop == 2) {
+          ; // skip the second line because we are logging the lastline and at 2 lastline is start
+        }
+        else {
+          interiorWaypoints.add(createTranslationWaypoint(lastline));
+        }
+        lastline = line;
+        loop++;
+      }
+      end = createPoseWaypoint(lastline);
+
+      return TrajectoryGenerator.generateTrajectory(
+            start,
+            interiorWaypoints,
+            end,
+            config);
+    }
+  }
+
+  /**
+   * Imports Path object and creates a trajectory with that info.
+   *
+   * @param filepath Full path to the file
+   * @param config TrajectoryConfig that sets the contraints for the Trajectory to be generated
+   *
+   * @return Trajectory created from the waypoints in the file.
+   */
+  public static Trajectory importPathToQuinticTrajectory(String filepath, TrajectoryConfig config) throws IOException {
+    try (FileReader fr = new FileReader(new File(filepath));
+      BufferedReader reader = new BufferedReader(fr)) {
+      
+      ControlVectorList controlVectors = new ControlVectorList();
+      int loop = 0;
+      String line;
+
+      while ((line = reader.readLine()) != null) {
+         if (loop == 0) {
+            if (!line.equals("X,Y,Tangent X,Tangent Y,Fixed Theta,Reversed,Name")) {
+              throw new RuntimeException("this isn’t a PathWeaver csv file");
+            }
+         }
+         else {
+            controlVectors.add(createControlVector(line));
+         }
+         loop++;
+      }
+      return TrajectoryGenerator.generateTrajectory(controlVectors,            
+            config);
+    }
+  }
+
+  private static Pose2d createPoseWaypoint(String input) {
+    String[] arrOfStr = input.split(",", 0);
+    // 8.21m is the Height of the field PathWeaver and traj use different starting points
```

Comment:
Is 8.21 always the height of the field?  Even for different fields?  Does this need to be configurable?

Reply:
good catch.  That was an adjustment for the current paths that use a field from a different starting spot.  The Pathweaver 206 PR I listed above adjusts for that.  The 8.21 shouldn't be here as it will be adjusted for in PathWeaver.  Unfortunately however this won't work for "old" waypoint files but I don't know a good way of getting around that.